dropwizard/src/main/java/com/linecorp/armeria/dropwizard/ArmeriaServerFactory.java

```diff
@@ -0,0 +1,335 @@
+     * @param accessLogWriter An instance of an {#link AccessLogWriter}
+     */
+    public void setAccessLogWriter(final @Valid AccessLogWriterFactory accessLogWriter) {
+        this.accessLogWriter = Objects.requireNonNull(
+                accessLogWriter, "server[type=\"" + TYPE + "\"].accessLogWriter");
+    }
+
+    public Size getMaxRequestLength() {
+        return maxRequestLength;
+    }
+
+    public void setMaxRequestLength(final Size maxRequestLength) {
+        this.maxRequestLength = maxRequestLength;
+    }
+
+    public int getMaxNumConnections() {
+        return maxNumConnections;
+    }
+
+    public void setMaxNumConnections(final int maxNumConnections) {
+        this.maxNumConnections = maxNumConnections;
+    }
+
+    @JsonIgnore
+    public ServerBuilder getServerBuilder() {
+        return serverBuilder;
+    }
+
+    @Override
+    public Server build(final Environment environment) {
+        Objects.requireNonNull(environment, "environment");
+        printBanner(environment.getName());
+        final MetricRegistry metrics = environment.metrics();
+        final ThreadPool threadPool = createThreadPool(metrics);
+        final Server server = buildServer(environment.lifecycle(), threadPool);
+
+        final JerseyEnvironment jersey = environment.jersey();
+        if (!isJerseyEnabled()) {
+            jersey.disable();
+        }
+
+        addDefaultHandlers(server, environment, metrics);
+        serverBuilder = getArmeriaServerBuilder(server, connector, metrics);
+        return server;
+    }
+
+    private void addDefaultHandlers(final Server server, final Environment environment,
+                                    final MetricRegistry metrics) {
+        final JerseyEnvironment jersey = environment.jersey();
+        final Handler applicationHandler = createAppServlet(
+                server,
+                jersey,
+                environment.getObjectMapper(),
+                environment.getValidator(),
+                environment.getApplicationContext(),
+                environment.getJerseyServletContainer(),
+                metrics);
+        final Handler adminHandler = createAdminServlet(
+                server,
+                environment.getAdminContext(),
+                metrics,
+                environment.healthChecks());
+        final ContextRoutingHandler routingHandler = new ContextRoutingHandler(ImmutableMap.of(
+                getApplicationContextPath(), applicationHandler,
+                getAdminContextPath(), adminHandler));
+        final Handler gzipHandler = buildGzipHandler(routingHandler);
+        server.setHandler(addStatsHandler(addRequestLog(server, gzipHandler, environment.getName())));
+    }
+
+    private ServerBuilder getArmeriaServerBuilder(final Server server,
+                                                  @Nullable final ConnectorFactory connector,
+                                                  @Nullable final MetricRegistry metricRegistry) {
```

Comment:
These two fields are not nullable.
Could you also remove the `final` keyword from the parameter declaration of all methods, please? 😄